spark/v3.4/spark/src/main/java/org/apache/iceberg/spark/actions/RewritePositionDeleteFilesSparkAction.java

```diff
@@ -133,16 +137,23 @@ public RewritePositionDeleteFiles.Result execute() {
   }
 
   private StructLikeMap<List<List<PositionDeletesScanTask>>> planFileGroups() {
-    CloseableIterable<PositionDeletesScanTask> fileTasks = planFiles();
+    Table deletesTable =
```

Comment:
Why add it here and why rename `fileTasks`? Don't we have to modify `planFiles` instead?

```
private CloseableIterable<PositionDeletesScanTask> planFiles() {
  Table deletesTable =
      MetadataTableUtils.createMetadataTableInstance(table, MetadataTableType.POSITION_DELETES);

  PositionDeletesBatchScan scan = (PositionDeletesBatchScan) deletesTable.newBatchScan();

  return CloseableIterable.transform(
      scan.baseTableFilter(filter).ignoreResiduals().planFiles(),
      task -> (PositionDeletesScanTask) task);
}
```